pkg/k8smgmt/appinst.go

```diff
@@ -408,17 +416,8 @@ func createOrUpdateAppInst(ctx context.Context, accessApi platform.AccessApi, cl
 	// for an AppInst must be stored in their own directory.
 
 	// Selector selects which objects to consider for pruning.
-	// Previously we used "--all", but that ends up deleting extra stuff,
-	// especially in the case of multi-tenant clusters. We want to
-	// transition to using the config label, but we can't use it for
-	// old configs that didn't have it. So we will have to continue
-	// to use "--all" for old stuff until those old configs eventually
-	// get removed naturally over time.
-	selector := "--all"
 	kconfArg := names.GetTenantKconfArg()
-	if names.MultitenantNamespace != "" {
-		selector = fmt.Sprintf("-l %s=%s", ConfigLabel, getConfigLabel(names))
-	}
+	selector := fmt.Sprintf("-l %s=%s", ConfigLabel, getConfigLabel(names))
```

Comment:
ah finally. This selector `--all` has been bugging me for a while now :)

Reply:
yes, this hit me during testing as well.